dev/planning-example.py

```diff
@@ -0,0 +1,292 @@
+
+# Define a target.
+t1 = FixedTarget(name='Polaris', ra='02:31:49.09', dec='+89:15:50.8')
+
+# initializing from astropy entities directly
+from astropy.coordinates import SkyCoord
+
+t1 = FixedTarget(name='Polaris',
+               coord=SkyCoord('02h31m49.09s', '+89d15m50.8s', frame='icrs'))
+
+# Leaves scope for NonSiderealTarget, etc.
+
+# Convenience methods for looking up/constructing targets by name via
+# astroquery?
+
+# Plot the airmass for the specified target:
+airmass_plot(t1, observer=obs, datetime=day)
+
+# ================================
+# Condition objects, observability
+# ================================
+
+"""
+Q: Can I observe this target on the night of May 1, 2015 between 18:30
+and 05:30 HST for 30 minutes somewhere between 15 degrees and 89 degrees
+altitude?
+"""
+
+# first we define the conditions
+from astroplan import Constraints, TimeWindow, AltitudeRange, AboveAirmass
+
+# times can be passed in as strings (interpreted as for get_date()) or
+# as astropy Time or datetime objects
+cons = [TimeWindow("2015-05-01 18:30", "2015-05-02 05:30"),
+        AltitudeRange(15.0*u.deg, 89.0*u.deg), AboveAirmass(1.2)]
+# define a target
+tgt = FixedTarget(name='S5', ra='14:20:00.00', dec='48:00:00.00')
+
+# returns an object with information about the observability with these
+# constraints
+info = obs.Constraints(tgt, cons)
+
+'''
+attributes of return object:
+  observable: a boolean -- whether target is observable at desired constraints
+  rise_time: a datetime -- first time target is visible in at these conditions
+  set_time: a datetime -- time target is no longer visible at these conditions
+
+Use of a return object means that additional items can be added in the
+future without breaking the API (e.g. subclasses are free to attach
+additional items).  It can be also useful to attach an "explanation" for why
+something cannot be observed.
+'''
+
+'''
+Suggest that Constraints implement a time range, an altitude range and an
+airmass limit at beginning.  Other things that could be added later:
+ - azimuth range
+ - airmass range
+
+Idea is that Constraints can be subclassed to add additional constraints.
+'''
+
+# ======================================================
+# Other useful calculations wrt an observer and a target
+#=======================================================
+
+# calculate the distance in alt and az degrees between two targets at
+# the given time (e.g. to calculate slew time)
+sf = FixedTarget(name='Sf', ra='09:40:00.00', dec='43:00:00.00')
+sm = FixedTarget(name='Sm', ra='10:30:00.00', dec='36:00:00.00')
+t = obs.get_date("2015-05-01 23:00")
+
+# Coordinate arithmetic gives separations in RA, Dec, alt, az
+dra, ddec = sf.ra - sm.ra, sf.dec - sm.dec
+dalt = obs.altaz(sf, t).alt - obs.altaz(sm, t).alt
+dazt = obs.altaz(sf, t).az - obs.altaz(sm, t).az
+
+# tell me about object sm in relation to observer obs at time t
+# returns an Observation object
+cr = obs.calc(sm, t)
```

Comment:
### Summary of comments on `calc`

**General**:
* @eteq : The name is unclear what it's calculating or why. 
* @jberlanga : maybe rename to `observe`, or `observability`?
* @cdeil : Let's try and find a better name for this than `CalculationResult`.
Maybe `Observation`?

**Pros**: 
* @ejeschke : This approach avoids a stateful approach like pyephem uses, which complicates the code using it excessively. The calculation result basically captures the relationship between a target, an observer and a time -- and allows you to calculate only those values that you need at your leisure. Suppose I want to evaluate 400 targets wrt 12 observing positions at time t1, and then just get the airmass and parallactic angle of these. With CalculationResult you just run this `obs.calc(tgt, t)` method the 400*12 times and extract `cr.airmass` and `cr.pang`. No extraneous calculations will be made, and nothing stateful is changed in the observer, target or time objects.
* @ejeschke : Convenience/preference: If I am an observer doing some simple scripting, I'd much rather write: `r = calc(sm, m, t); r.airmass; r.pang` than `am = airmass(obs, sm, t); pang = parallactic_angle(obs, sm, t)`
* @ejeschke : If we want to develop for interactive users then we need to provide convenient shorthand ways of expressing things or else we leave it to someone else to write that layer
 
**Cons**: 
* @eteq : "I'm not a big fan of a 'do everything' sort of method."
* @eteq : I still think it's important that users be able to provide their own sorts of things to calculate within the framework, rather than forcing them to only have what we give them.  I was going to propose that we design an API for how users could add new things to calculate, but as I thought about it it always ended up basically just "provide a function to do the calculation". So that makes me think it's better just to use the function itself.